PROTOCOL.md

```diff
@@ -2,179 +2,287 @@
 
 Extreme portability is a key design goal of libchan.
 
-This document specifies the libchan protocol to allow multiple implementations to co-exist with
-full interoperability.
+This document specifies the libchan protocol to allow multiple implementations
+to co-exist with full interoperability.
 
 ## Version
 
-No version yet.
+0.2.0
 
-## Author
+## Authors
 
 Solomon Hykes <solomon@docker.com>
+Derek McGowan <derek@docker.com>
+Matteo Collina <matteo.collina@gmail.com>
 
 ## Status
 
-This specification is still work in progress. Things will change, probably in reverse-incompatible ways.
-We hope to reach full API stability soon.
+This specification is nearing a stable release. The protocol still may change in
+reverse-incompatible ways.
 
 ## Terminology
 
 ### Channel
 
-A `channel` is an object which allows 2 concurrent programs to communicate with each other. The semantics
-of a libchan channel are very similar (but not identical) to those of Go's native channels.
+A `channel` is an object which allows 2 concurrent programs to communicate with
+each other. The semantics of a libchan channel are very similar (but not
+identical) to those of Go's native channels. A channel may be used
+synchronously, but do not support Go's channel select semantics.
 
-A channel has 2 ends: a `Sender` end and a `Receiver` end. The Sender can send messages and close the channel.
-The Receiver can receive messages. Messages arrive in the same order they were sent.
+A channel has 2 ends: a `Sender` end and a `Receiver` end. The Sender can send
+messages and close the channel. The Receiver can receive messages. Messages
+arrive in the same order they were sent.
 
-A channel is uni-directional: messages can only flow in one direction. So channels are more similar to pipes
-than to sockets.
+A channel is uni-directional: messages can only flow in one direction. So
+channels are more similar to pipes than to sockets.
 
 ### Message
 
-A message is a discrete packet of data which can be sent on a channel. Messages are structured into multiple
-fields. The protocol defines which data types can be carried by a message, and how transports should encode and
-decode them.
+A message is a discrete packet of data which can be sent on a channel. The
+protocol defines which data types can be carried sent as a message, and how
+transports should encode and decode them.
```

Comment:
I would a brief explanation of what a message is, e.g. a JSON map with custom types.